From a89e1e448c4ac4fb019d6144f4db7b9ebf3c7e48 Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Tue, 29 Apr 2025 21:28:38 +0300
Subject: [PATCH] [SW-222861] g3: pldm: mask pdrs by platform

Some PDRs should be supported on HL325 and not HL338.
Mask the following:
	Gaudi3 OAM power sensor: ADC 48V input
	Gaudi3 OAM voltage sensor: 48V OAM Input ADC
	Gaudi3 OAM voltage sensor: B2B PVREF
	Gaudi3 current power consumption, 54V domain
	Gaudi3 Highest consumed power, 54V domain
	Gaudi3 power effecter: Limit maximal supported power envelope
	Gaudi3 energy consumed by the 54V PSU
	Gaudi3 Peak power consumption observed in 1 minute time window
	Gaudi3 Last power reduction duration

Change-Id: I53a49da61360a7eb33a981f778e4bf559e5425c4
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/arch/gaudi3/ras/pldm/pldm_pdr.c b/arch/gaudi3/ras/pldm/pldm_pdr.c
index 5e78e68..68c27a0 100644
--- a/arch/gaudi3/ras/pldm/pldm_pdr.c
+++ b/arch/gaudi3/ras/pldm/pldm_pdr.c
@@ -553,22 +553,27 @@
 			},
 		},
 	}},
-	[PLDM_RIDX_SENSOR_CURRENT_ADC_48V_POWER] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_CURRENT_ADC_48V_POWER,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+	[PLDM_RIDX_SENSOR_CURRENT_ADC_48V_POWER] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
 		},
-		.sensor_id = PLDM_SID_CURRENT_ADC_48V_POWER,
-		.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
-		.unit_modifier = -3,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.sensor_name = "currentADC48Vpower",
-	}},
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_CURRENT_ADC_48V_POWER,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.sensor_id = PLDM_SID_CURRENT_ADC_48V_POWER,
+			.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
+			.unit_modifier = -3,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.sensor_name = "currentADC48Vpower",
+		},
+	},
 	[PLDM_RIDX_SENSOR_CURRENT_ADC_12V_POWER] = { .compact_numeric_sensor = {
 		.header = {
 			.record_handle = PLDM_RH_SENSOR_CURRENT_ADC_12V_POWER,
@@ -665,48 +670,53 @@
 		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
 		.sensor_name = "historicalVRMTemperature",
 	}},
-	[PLDM_RIDX_SENSOR_OAM_INPUT_ADC_48V] = { .numeric_sensor_u32 = {
-		.common = {
-			.header = {
-				.record_handle = PLDM_RH_SENSOR_OAM_INPUT_ADC_48V,
-				.header_version = PLDM_PDR_HEADER_VERSION,
-				.type = PLDM_PDR_TYPE_NUMERIC_SENSOR,
-				.record_change_number = 0,
+	[PLDM_RIDX_SENSOR_OAM_INPUT_ADC_48V] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
+		},
+		.numeric_sensor_u32 = {
+			.common = {
+				.header = {
+					.record_handle = PLDM_RH_SENSOR_OAM_INPUT_ADC_48V,
+					.header_version = PLDM_PDR_HEADER_VERSION,
+					.type = PLDM_PDR_TYPE_NUMERIC_SENSOR,
+					.record_change_number = 0,
+				},
+				.terminus_handle = 0,
+				.sensor_id = PLDM_SID_OAM_INPUT_ADC_48V,
+				.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+				.entity_instance_number = 1,
+				.container_id = PLDM_CONTAINER_ADDIN_CARD,
+				.init = PLDM_PDR_INIT_NONE,
+				.aux_names_pdr = true,
+				.base_unit = PLDM_PDR_SENSOR_UNIT_VOLTS,
+				.unit_modifier = -3,
+				.is_linear = true,
+				.data_size = PLDM_DATA_SIZE_UINT32,
+				.resolution = 1,
 			},
-			.terminus_handle = 0,
-			.sensor_id = PLDM_SID_OAM_INPUT_ADC_48V,
-			.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
-			.entity_instance_number = 1,
-			.container_id = PLDM_CONTAINER_ADDIN_CARD,
-			.init = PLDM_PDR_INIT_NONE,
-			.aux_names_pdr = true,
-			.base_unit = PLDM_PDR_SENSOR_UNIT_VOLTS,
-			.unit_modifier = -3,
-			.is_linear = true,
-			.data_size = PLDM_DATA_SIZE_UINT32,
-			.resolution = 1,
+			.volatility = {
+				.restart = 1,
+				.power_up = 1,
+				.hard_reset = 1,
+				.warm_reset = 1,
+				.terminus_online = 1,
+			},
+			.state_transition_interval = 0.0,
+			.update_interval = 120.0,
+			.max_readable = UINT_MAX,
+			.min_readable = 0,
+			.range_format = PLDM_DATA_SIZE_UINT32,
+			.range_field_support = {
+				.nominal_value = 1,
+				.normal_max = 1,
+				.normal_min = 1,
+			},
+			.nominal_value = 54000,
+			.normal_max = 60000,
+			.normal_min = 44000,
 		},
-		.volatility = {
-			.restart = 1,
-			.power_up = 1,
-			.hard_reset = 1,
-			.warm_reset = 1,
-			.terminus_online = 1,
-		},
-		.state_transition_interval = 0.0,
-		.update_interval = 120.0,
-		.max_readable = UINT_MAX,
-		.min_readable = 0,
-		.range_format = PLDM_DATA_SIZE_UINT32,
-		.range_field_support = {
-			.nominal_value = 1,
-			.normal_max = 1,
-			.normal_min = 1,
-		},
-		.nominal_value = 54000,
-		.normal_max = 60000,
-		.normal_min = 44000,
-	}},
+	},
 	[PLDM_RIDX_SENSOR_OAM_INPUT_ADC_12V] = { .numeric_sensor_u32 = {
 		.common = {
 			.header = {
@@ -791,49 +801,53 @@
 		.normal_max = 3465,
 		.normal_min = 3135,
 	}},
-	[PLDM_RIDX_SENSOR_B2B_PVREF] = { .numeric_sensor_u32 = {
-		.common = {
-			.header = {
-				.record_handle = PLDM_RH_SENSOR_B2B_PVREF,
-				.header_version = PLDM_PDR_HEADER_VERSION,
-				.type = PLDM_PDR_TYPE_NUMERIC_SENSOR,
-				.record_change_number = 0,
+	[PLDM_RIDX_SENSOR_B2B_PVREF] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
+		},
+		.numeric_sensor_u32 = {
+			.common = {
+				.header = {
+					.record_handle = PLDM_RH_SENSOR_B2B_PVREF,
+					.header_version = PLDM_PDR_HEADER_VERSION,
+					.type = PLDM_PDR_TYPE_NUMERIC_SENSOR,
+					.record_change_number = 0,
+				},
+				.terminus_handle = 0,
+				.sensor_id = PLDM_SID_B2B_PVREF,
+				.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+				.entity_instance_number = 1,
+				.container_id = PLDM_CONTAINER_ADDIN_CARD,
+				.init = PLDM_PDR_INIT_NONE,
+				.aux_names_pdr = true,
+				.base_unit = PLDM_PDR_SENSOR_UNIT_VOLTS,
+				.unit_modifier = -3,
+				.is_linear = true,
+				.data_size = PLDM_DATA_SIZE_UINT32,
+				.resolution = 1,
 			},
-			.terminus_handle = 0,
-			.sensor_id = PLDM_SID_B2B_PVREF,
-			.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
-			.entity_instance_number = 1,
-			.container_id = PLDM_CONTAINER_ADDIN_CARD,
-			.init = PLDM_PDR_INIT_NONE,
-			.aux_names_pdr = true,
-			.base_unit = PLDM_PDR_SENSOR_UNIT_VOLTS,
-			.unit_modifier = -3,
-			.is_linear = true,
-			.data_size = PLDM_DATA_SIZE_UINT32,
-			.resolution = 1,
+			.volatility = {
+				.restart = 1,
+				.power_up = 1,
+				.hard_reset = 1,
+				.warm_reset = 1,
+				.terminus_online = 1,
+			},
+			.state_transition_interval = 0.0,
+			.update_interval = 120.0,
+			.max_readable = UINT_MAX,
+			.min_readable = 0,
+			.range_format = PLDM_DATA_SIZE_UINT32,
+			.range_field_support = {
+				.nominal_value = 1,
+				.normal_max = 1,
+				.normal_min = 1,
+			},
+			.nominal_value = 1200,
+			.normal_max = 1260,
+			.normal_min = 1140,
 		},
-		.volatility = {
-			.restart = 1,
-			.power_up = 1,
-			.hard_reset = 1,
-			.warm_reset = 1,
-			.terminus_online = 1,
-		},
-		.state_transition_interval = 0.0,
-		.update_interval = 120.0,
-		.max_readable = UINT_MAX,
-		.min_readable = 0,
-		.range_format = PLDM_DATA_SIZE_UINT32,
-		.range_field_support = {
-			.nominal_value = 1,
-			.normal_max = 1,
-			.normal_min = 1,
-		},
-		.nominal_value = 1200,
-		.normal_max = 1260,
-		.normal_min = 1140,
-	}},
-
+	},
 	[PLDM_RIDX_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_MAIN] = { .state_sensor = {
 		.header = {
 			.record_handle = PLDM_RH_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_MAIN,
@@ -2799,40 +2813,50 @@
 			},
 		},
 	}},
-	[PLDM_RIDX_SENSOR_CURRENT_SOC_54V_POWER] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_CURRENT_SOC_54V_POWER,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+	[PLDM_RIDX_SENSOR_CURRENT_SOC_54V_POWER] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
 		},
-		.terminus_handle = 0,
-		.sensor_id = PLDM_SID_CURRENT_SOC_54V_POWER,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
-		.unit_modifier = -3,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.sensor_name = "currentSOC54VPower",
-	}},
-	[PLDM_RIDX_SENSOR_PEAK_SOC_54V_POWER] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_PEAK_SOC_54V_POWER,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_CURRENT_SOC_54V_POWER,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.sensor_id = PLDM_SID_CURRENT_SOC_54V_POWER,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
+			.unit_modifier = -3,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.sensor_name = "currentSOC54VPower",
 		},
-		.terminus_handle = 0,
-		.sensor_id = PLDM_SID_PEAK_SOC_54V_POWER,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
-		.unit_modifier = -3,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.sensor_name = "peakSOC54VPower",
-	}},
+	},
+	[PLDM_RIDX_SENSOR_PEAK_SOC_54V_POWER] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
+		},
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_PEAK_SOC_54V_POWER,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.sensor_id = PLDM_SID_PEAK_SOC_54V_POWER,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
+			.unit_modifier = -3,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.sensor_name = "peakSOC54VPower",
+		},
+	},
 	[PLDM_RIDX_SENSOR_CURRENT_SOC_12V_POWER] = { .compact_numeric_sensor = {
 		.header = {
 			.record_handle = PLDM_RH_SENSOR_CURRENT_SOC_12V_POWER,
@@ -2867,23 +2891,28 @@
 		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
 		.sensor_name = "peakSOC12VPower",
 	}},
-	[PLDM_RIDX_SENSOR_LAST_POWER_REDUCTION_DURATION] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_LAST_POWER_REDUCTION_DURATION,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+	[PLDM_RIDX_SENSOR_LAST_POWER_REDUCTION_DURATION] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
 		},
-		.terminus_handle = 0,
-		.sensor_id = PLDM_SID_LAST_POWER_REDUCTION_DURATION,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_SECONDS,
-		.unit_modifier = -6,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.sensor_name = "lastPowerReductionDuration",
-	}},
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_LAST_POWER_REDUCTION_DURATION,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.sensor_id = PLDM_SID_LAST_POWER_REDUCTION_DURATION,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_SECONDS,
+			.unit_modifier = -6,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.sensor_name = "lastPowerReductionDuration",
+		},
+	},
 	[PLDM_RIDX_SENSOR_LAST_POWER_BREAK_DURATION] = { .compact_numeric_sensor = {
 		.header = {
 			.record_handle = PLDM_RH_SENSOR_LAST_POWER_BREAK_DURATION,
@@ -2952,63 +2981,78 @@
 		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
 		.sensor_name = "hbmThrottlingDuration",
 	}},
-	[PLDM_RIDX_EFFECTER_LIMIT_TDP_54V] = { .numeric_effecter = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_LIMIT_TDP_54V,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_NUMERIC_EFFECTER,
-			.record_change_number = 0,
+	[PLDM_RIDX_EFFECTER_LIMIT_TDP_54V] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
 		},
-		.effecter_id = PLDM_SID_LIMIT_TDP_54V,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.aux_names = true,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
-		.rate_unit = PLDM_PDR_SENSOR_RATE_NONE,
-		.data_size = PLDM_DATA_SIZE_UINT32,
-		.max_settable = 850,
-		.min_settable = 150,
-		.range_format = PLDM_DATA_SIZE_UINT32,
-		.range_field_support = {
-			.normal_max = 1,
+		.numeric_effecter = {
+			.header = {
+				.record_handle = PLDM_RH_EFFECTER_LIMIT_TDP_54V,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_NUMERIC_EFFECTER,
+				.record_change_number = 0,
+			},
+			.effecter_id = PLDM_SID_LIMIT_TDP_54V,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.aux_names = true,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_WATTS,
+			.rate_unit = PLDM_PDR_SENSOR_RATE_NONE,
+			.data_size = PLDM_DATA_SIZE_UINT32,
+			.max_settable = 850,
+			.min_settable = 150,
+			.range_format = PLDM_DATA_SIZE_UINT32,
+			.range_field_support = {
+				.normal_max = 1,
+			},
+			.normal_max = 850,
 		},
-		.normal_max = 850,
-	}},
-	[PLDM_RIDX_SENSOR_ENERGY_CONSUMED_54V_JOULES] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_ENERGY_CONSUMED_54V_JOULES,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+	},
+	[PLDM_RIDX_SENSOR_ENERGY_CONSUMED_54V_JOULES] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
 		},
-		.terminus_handle = 0,
-		.sensor_id = PLDM_SID_ENERGY_CONSUMED_54V_JOULES,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_JOULES,
-		.unit_modifier = -3,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.sensor_name = "energyConsumed54VJoules",
-	}},
-	[PLDM_RIDX_SENSOR_PEAK_POWER_CONSUMPTION_54V_JOULES] = { .compact_numeric_sensor = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_PEAK_POWER_CONSUMPTION_54V_JOULES,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
-			.record_change_number = 0,
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_ENERGY_CONSUMED_54V_JOULES,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.sensor_id = PLDM_SID_ENERGY_CONSUMED_54V_JOULES,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_JOULES,
+			.unit_modifier = -3,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.sensor_name = "energyConsumed54VJoules",
 		},
-		.terminus_handle = 0,
-		.sensor_id = PLDM_SID_PEAK_POWER_CONSUMPTION_54V_JOULES,
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance_number = 1,
-		.container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.base_unit = PLDM_PDR_SENSOR_UNIT_JOULES,
-		.unit_modifier = -3,
-		.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
-		.sensor_name = "peakPowerConsumption54VJoules",
-	}},
+	},
+	[PLDM_RIDX_SENSOR_PEAK_POWER_CONSUMPTION_54V_JOULES] = {
+		.meta = {
+			.unsupported_platform_mask = PLDM_PLATFORM_MASK_HL338,
+		},
+		.compact_numeric_sensor = {
+			.header = {
+				.record_handle = PLDM_RH_SENSOR_PEAK_POWER_CONSUMPTION_54V_JOULES,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_COMPACT_NUMERIC_SENSOR,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.sensor_id = PLDM_SID_PEAK_POWER_CONSUMPTION_54V_JOULES,
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance_number = 1,
+			.container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.base_unit = PLDM_PDR_SENSOR_UNIT_JOULES,
+			.unit_modifier = -3,
+			.occurrence_rate = PLDM_PDR_SENSOR_RATE_NONE,
+			.sensor_name = "peakPowerConsumption54VJoules",
+		},
+	},
 	[PLDM_RIDX_SENSOR_COMPOSITE_TEMPERATURE] = { .compact_numeric_sensor = {
 		.header = {
 			.record_handle = PLDM_RH_SENSOR_COMPOSITE_TEMPERATURE,
